src/messages/MessageBuilder.cpp

```diff
@@ -2746,6 +2757,28 @@ void MessageBuilder::appendTwitchBadges(const QVariantMap &tags,
         return;
     }
 
+    if (this->message().flags.has(MessageFlag::SharedMessage))
+    {
+        QString sourceId = tags["source-room-id"].toString();
+        std::optional<QString> sourceName;
+        if (twitchChannel->roomId() == sourceId)
+        {
+            sourceName = twitchChannel->getName();
+        }
+        else
+        {
+            sourceName =
+                getApp()->getTwitch()->getOrPopulateChannelCache(sourceId);
+        }
+
+        if (sourceName.has_value())
+        {
+            this->emplace<BadgeElement>(makeSharedChatBadge(sourceName.value()),
+                                        MessageElementFlag::BadgeSharedChannel)
+                ->setLink({Link::UserInfo, sourceName.value()});
+        }
```

Comment:
Since `sourceName` can also be empty, we should check for both has_value and `sourceName.value().isEmpty()`

Reply:
`makeSharedChatBadge` has that check & can intentionally be passed an empty string

but yeah that can be problematic for the UserInfo link

Reply:
Done in https://github.com/Chatterino/chatterino2/pull/5661/commits/021f1a0bfc7c77c18aaa7f65a62f7263b7db6893